export tx provider from sdk & remove safe util from infra

nbayindirli opened the pull request:
### Description

* Exports new tx provider + Gnosis safe util from SDK
* Removes the Gnosis safe util from infra

### Drive-by changes

* Reminder to add a changeset
  * Output:
```
➜  hyperlane-monorepo-official git:(noah/move-safe) ✗ git commit -m "add changeset reminder to husky script"
  → No staged files match any configured task.
  📝 If you haven't yet, please add a changeset for your changes via 'yarn changeset'
  [noah/move-safe 3e668d0a7] add changeset reminder to husky script
   1 file changed, 2 insertions(+)
```

### Related issues

* Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3684

### Backward compatibility

* Yes

### Testing

* n/a

jmrossy reviewed (commented):
Seems fine to be but looks like lots of CIs aren't happy.
Don't forget to add a changeset :)

nbayindirli commented:
> Seems fine to be but looks like lots of CIs aren't happy. Don't forget to add a changeset :)

Added! CIs also passing now. Also added reminder to add changesets

jmrossy reviewed (approved):
🚢
